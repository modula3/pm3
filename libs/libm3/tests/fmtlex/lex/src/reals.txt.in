# Last modified on Wed Mar 23 15:44:05 PST 1994 by heydon

# This file is used to measure the performance of the lexer on real numbers.

# No exponent
r 0.12345abc
r 12.345abc
r 1234.5abc

# Exponent
r 1.2345e-1abc
r 1.2345e1abc
r 1.2345e12abc

# Lots of digits
r 0.00000000000000000001234567890123456789abc
r 0.00000000012345678901234567890123456789abc
r 1.23456789012345678901234567890123456789abc
r 123456789.01234567890123456789012345678901234567890123456789abc
r 1234567890123456789.0123456789012345678901234567890123456789abc
r 12345678901234567890123456789.012345678901234567890123456789abc

# Trigger EOF exception
r 123
#r 123
r 123a
#r 1.23
#r 1.23e2

