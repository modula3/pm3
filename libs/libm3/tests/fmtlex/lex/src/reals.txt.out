r 0.12345abc
-> 0.12345
-> "abc"

r 12.345abc
-> 12.345
-> "abc"

r 1234.5abc
-> 1234.5
-> "abc"

r 1.2345e-1abc
-> 0.12345
-> "abc"

r 1.2345e1abc
-> 12.345
-> "abc"

r 1.2345e12abc
-> 1.2345e12
-> "abc"

r 0.00000000000000000001234567890123456789abc
-> 1.2345679e-20
-> "abc"

r 0.00000000012345678901234567890123456789abc
-> 1.2345679e-10
-> "abc"

r 1.23456789012345678901234567890123456789abc
-> 1.2345679
-> "abc"

r 123456789.01234567890123456789012345678901234567890123456789abc
-> 1.2345679e8
-> "abc"

r 1234567890123456789.0123456789012345678901234567890123456789abc
-> 1.234568e18
-> "abc"

r 12345678901234567890123456789.012345678901234567890123456789abc
-> 1.2345679e28
-> "abc"

r 123
-> 123
-> ""

r 123a
-> 123
-> "a"

